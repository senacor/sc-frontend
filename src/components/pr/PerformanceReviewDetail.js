import React, { useContext, useEffect, useState } from 'react';
import CircularProgress from '@material-ui/core/CircularProgress/index';
import PrState from './PrState';
import PrTabs from './PrTabs';
import PrDetailInformation from './PrDetailInformation';
import { fetchMeeting } from '../../calls/meetings';
import { fetchPrById } from '../../calls/pr';
import { ErrorContext, MeetingContext, PrContext } from '../App';

const PerformanceReviewDetail = props => {
  const { value: pr, setValue: setPr } = useContext(PrContext.context);
  const [isLoading, setIsLoading] = useState({});
  const errorContext = useContext(ErrorContext.context);
  const { value: meeting, setValue: setMeeting } = useContext(
    MeetingContext.context
  );

  useEffect(() => {
    const afterPrFetched = pr => {
      setPr(pr);
      fetchMeeting(pr, setMeeting, errorContext);
    };
    fetchPrById(
      props.match.params.id,
      afterPrFetched,
      setIsLoading,
      errorContext
    );
  }, []);

  if (isLoading || !pr || Object.entries(pr).length === 0) {
    return <CircularProgress />;
  }

  return (
    <React.Fragment>
      {pr && <PrDetailInformation pr={pr} meeting={meeting} />}
      {pr && <PrState prById={pr} />}
      {pr && <PrTabs pr={pr} meeting={meeting} />}
    </React.Fragment>
  );
};

export default PerformanceReviewDetail;
