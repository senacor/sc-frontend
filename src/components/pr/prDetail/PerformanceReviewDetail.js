import React, { Component } from 'react';
import PrState from './PrState';
import PrTabs from './PrTabs';
import { connect } from 'react-redux';
import { getPrDetail } from '../../../reducers/selector';
import * as actions from '../../../actions';
import withLoading from '../../hoc/Loading';

export class PerformanceReviewDetail extends Component {
  render() {
    let { prById } = this.props;
    return (
      <React.Fragment>
        <PrState />
        {prById ? <PrTabs /> : null}
      </React.Fragment>
    );
  }
}

PerformanceReviewDetail.propTypes = {};

export default connect(
  state => ({
    prById: getPrDetail()(state),
    isLoading: state.isLoading
  }),
  {
    fetchPrById: actions.fetchPrById,
    fetchMeeting: actions.fetchMeeting
  }
)(
  withLoading(props => {
    return props
      .fetchPrById(props.match.params.id)
      .then(pr => props.fetchMeeting(pr));
  })(PerformanceReviewDetail)
);
