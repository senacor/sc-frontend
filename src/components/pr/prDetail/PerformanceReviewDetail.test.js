import React from 'react';
import { PerformanceReviewDetail } from './PerformanceReviewDetail';
import { shallow } from 'enzyme';

describe('PerformanceReviewDetail Component', () => {
  it('should match snapshot', () => {
    const component = shallow(<PerformanceReviewDetail />);
    expect(component).toMatchSnapshot();
  });
});
