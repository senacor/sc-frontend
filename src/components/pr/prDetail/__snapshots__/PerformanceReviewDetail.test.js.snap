// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerformanceReviewDetail Component should match snapshot 1`] = `
<React.Fragment>
  <Connect(WithLoading(WithStyles(PrState))) />
  <WithStyles(PrTabs) />
</React.Fragment>
`;
